polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -436,8 +437,10 @@ static const clang::ConstantArrayType *getCAT(const clang::Type *T) {
 
 ValueCategory MLIRScanner::VisitArrayInitLoop(clang::ArrayInitLoopExpr *expr,
                                               ValueCategory tostore) {
+  if (!SuppressWarnings)
```

Comment:
Do you think it make sense to have a function like:
```
void warn(const Twine &msg) {
  if (SuppressWarnings)
    return;

  llvm::WithColor::warning() << msg;
}
```
and replace all these warnings to calling this function?